common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -496,6 +499,51 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
         PlatformDependent0.copyMemory(null, srcAddr, dst, BYTE_ARRAY_BASE_OFFSET + dstIndex, length);
     }
 
+    public static ByteBuffer allocateDirectNoCleaner(int capacity) {
+        if (DIRECT_MEMORY_COUNTER != null) {
+            for (;;) {
+                long used = DIRECT_MEMORY_COUNTER.get();
+                long expect = used + capacity;
+                if (expect > maxDirectMemory()) {
+                    throw new OutOfMemoryError("No more memory available");
+                }
+                if (DIRECT_MEMORY_COUNTER.compareAndSet(used, expect)) {
+                    break;
+                }
+            }
+        }
+        try {
+            return PlatformDependent0.allocateDirectNoCleaner(capacity);
+        } catch (OutOfMemoryError e) {
```

Comment:
catch `Throwable`? We should always decrement `DIRECT_MEMORY_COUNTER` if an exception is thrown right?